### PR TITLE
Update `smol_str`

### PR DESCRIPTION
**Pull request by BenjaminBrienen**
- [x] Tested on all platforms changed
- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

`smol_str` is part of the crate's API, which causes type mismatches that can't be resolved downstream.

### CODE REVIEW

**Comment by kchibisov** [comment]
Unless you use winit master this won't solve anything, and you can use as many crates as you want and cast things if you use newer version of smol_str.

I'm not against updating it now, I just want to get rid of it entirely in 0.31, and this can not be backported.

I don't mind temporary bumping on master though if someone needs it.

**Comment by BenjaminBrienen** [comment]
Well it has to be updated in master before it gets released, right? It's fine if it takes time. This is just the step to get the ball rolling.

**Comment by kchibisov** [comment]
Sure, I just tend to update all the deps closer to release date, so you don't update more than once during cycle.

File: src/changelog/unreleased.md [hunk 1]
```diff
@@ -156,6 +156,7 @@ changelog entry.
     identify a finger in a multi-touch interaction. Replaces the old `Touch::id`.
```
Comment (aloucks) [inline src/changelog/unreleased.md:156]: The indentation is wrong.

File: src/changelog/unreleased.md [hunk 2]
```diff
@@ -156,6 +156,7 @@ changelog entry.
...
   - In the same spirit rename `DeviceEvent::MouseMotion` to `PointerMotion`.
   - Remove `Force::Calibrated::altitude_angle`.
+- Update `smol_str` to version `0.3`
```
Comment (aloucks) [inline src/changelog/unreleased.md:159]: The indentation is wrong.

Comment (BenjaminBrienen) [reply src/changelog/unreleased.md:159]: How so?
image

Comment (aloucks) [reply src/changelog/unreleased.md:159]: My mistake! I didn't check the context that it was in.

**Comment by madsmtm** [comment]
Thanks!

**Pull request merged**